code/extensions/DMSDocumentCartExtension.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+/**
+ * Class DMSDocumentCartExtension
+ *
+ * @property Boolean     AllowedInCart
+ * @property Int         PrintRequestCount
+ *
+ * @property DMSDocument owner
+ */
+class DMSDocumentCartExtension extends DataExtension
+{
+    /**
+     * @var DMSDocumentCartController
+     */
+    private $cartController;
+
+    private static $db = array(
+        'AllowedInCart'     => 'Boolean',
+        // Running total of print requests on this document
+        'PrintRequestCount' => 'Int',
+    );
+
+    /**
+     * Returns if a Document is permitted to reflect in a cart
+     *
+     * @return boolean
+     */
+    public function isAllowedInCart()
+    {
+        return $this->owner->AllowedInCart;
+    }
+
+    public function updateCMSFields(FieldList $fields)
+    {
+        $fields->insertBefore(CheckboxField::create(
+            'AllowedInCart',
+            _t('DMSDocument.ALLOWED_IN_CART', 'Allowed in document cart')
+        ), 'Description');
+    }
+
+    /**
+     * Increments the number of times a document was printed
+     *
+     * @return DMSDocument
+     */
+    public function incrementPrintRequest()
+    {
+        $this->owner->PrintRequestCount++;
+        $this->owner->write();
+
+        return $this->owner;
+    }
+
+    /**
+     * Checks if a given document already exists within the Cart. True if it does, false otherwise
+     *
+     * @return bool
+     */
+    public function isInCart()
+    {
+        return $this->getCart()->isInCart($this->owner->ID) ? true : false;
+    }
+
+    /**
+     * Checks to see if this extension (DMSDocumentCartExtension) has been loaded.
+     *
+     * @return bool
+     */
+    public function HasCartExtension()
+    {
+        return $this->owner->hasExtension('DMSDocumentCartExtension');
```

Comment:
This is a bit of an oxy-moron - the method will return true if the extension is applied or throw and error if it doesn't. I think this kind of functionality is best left to `DMSDocument::has_extension('DMSDocumentCartExtension')` instead - thoughts?

Otherwise the method should be camelCased